### ISSUE TITLE
Bundler 1.8 doesn't recognize mirror config from .bundle/config generated by bundler < 1.8

### ISSUE THREAD

**Issue by soupmatt** [bundler/bundler#3458]
I'm seeing some odd things with how bundler acts with mirror configs.

ruby version: 2.2.1
rubygems version: 2.4.3

Here's my `~/.bundle/config` that I generated with an earlier version of bundler:
```yaml
---
BUNDLE_WITHOUT: linux:sqlserver
BUNDLE_MIRROR__HTTPS://RUBYGEMS.ORG: http://rubygems.dev
BUNDLE_LOCAL__RAILS_ADMIN-PUNDIT: "/Users/mcampbell/dev/opensource/rails_admin-pundit"
BUNDLE_LOCAL__RAILS_ADMIN: "/Users/mcampbell/dev/opensource/rails_admin"
```

Under bundler 1.7.13, bundler recognizes the mirror config correctly:
```
➜ bundle --version
Bundler version 1.7.13
➜ bundle config
Settings are listed in order of priority. The top value will be used.
without
Set for the current user (/Users/mcampbell/.bundle/config): "linux:sqlserver"

mirror.https://rubygems.org
Set for the current user (/Users/mcampbell/.bundle/config): "http://rubygems.dev"

local.rails_admin-pundit
Set for the current user (/Users/mcampbell/.bundle/config): "/Users/mcampbell/dev/opensource/rails_admin-pundit"

local.rails_admin
Set for the current user (/Users/mcampbell/.bundle/config): "/Users/mcampbell/dev/opensource/rails_admin"
```

Under bundler 1.8.3, it does not recognize the old mirror config.
```
➜ bundle --version
Bundler version 1.8.3
➜ bundle config
Settings are listed in order of priority. The top value will be used.
without
Set for the current user (/Users/mcampbell/.bundle/config): "linux:sqlserver"

mirror.https://rubygems.org
You have not configured a value for `mirror.https://rubygems.org`

local.rails_admin-pundit
Set for the current user (/Users/mcampbell/.bundle/config): "/Users/mcampbell/dev/opensource/rails_admin-pundit"

local.rails_admin
Set for the current user (/Users/mcampbell/.bundle/config): "/Users/mcampbell/dev/opensource/rails_admin"
```

If I then run `bundle config mirror.https://rubygems.org http://rubygems.dev`, which I copied and edited from the help given by `bundle help config`, it changes my `~/.bundle/config` file to this:
```yaml
---
BUNDLE_WITHOUT: linux:sqlserver
BUNDLE_MIRROR__HTTPS://RUBYGEMS.ORG: http://rubygems.dev
BUNDLE_LOCAL__RAILS_ADMIN-PUNDIT: "/Users/mcampbell/dev/opensource/rails_admin-pundit"
BUNDLE_LOCAL__RAILS_ADMIN: "/Users/mcampbell/dev/opensource/rails_admin"
BUNDLE_LOCAL__SAML_IDP: "/Users/mcampbell/dev/vibes/saml_idp"
BUNDLE_MIRROR__HTTPS://RUBYGEMS__ORG/: http://rubygems.dev
```

It looks like it doesn't like the lack of a trailing slash on the old config value. And once I make this change, bundler 1.7.13 complains about the entry with the trailing slash.
```
➜ bundle --version
Bundler version 1.7.13
➜ bundle config
Settings are listed in order of priority. The top value will be used.
without
Set for the current user (/Users/mcampbell/.bundle/config): "linux:sqlserver"

mirror.https://rubygems.org
Set for the current user (/Users/mcampbell/.bundle/config): "http://rubygems.dev"

local.rails_admin-pundit
Set for the current user (/Users/mcampbell/.bundle/config): "/Users/mcampbell/dev/opensource/rails_admin-pundit"

local.rails_admin
Set for the current user (/Users/mcampbell/.bundle/config): "/Users/mcampbell/dev/opensource/rails_admin"

mirror.https://rubygems.org/
You have not configured a value for `mirror.https://rubygems.org/`
```

In conclusion, I have to have both keys in my `~/.bundle/config` file in order for the mirror feature to work under bundler 1.8 and bundler 1.7.

**Comment by indirect** [comment]
ugh, that's no good :( thanks for pointing this out.

**Comment by indirect** [comment]
I've just written a test specifically to make sure that Bundler 1.8 will read and use the mirror settings written by Bundler 1.7, in commit ad2e4530b55709034ed11d96765982e4b8d34e5c. If you're not seeing Bundler 1.8 able to honor the mirror setting that you set with 1.7, please provide everything in [ISSUES](https://github.com/bundler/bundler/blob/master/ISSUES.md) and we'll try to troubleshoot.

While semver promises that future versions will honor the settings of previous versions, it does not promise that previous versions will be able to read settings written by future versions—please upgrade to Bundler 1.8 everywhere if you need to read the mirror setting written by 1.8.
